src/Messages/Messages.php

```diff
@@ -0,0 +1,281 @@
+<?php
+
+/**
+ * This file is part of the Phalcon Framework.
+ *
+ * (c) Phalcon Team <team@phalcon.io>
+ *
+ * For the full copyright and license information, please view the LICENSE.txt
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Phalcon\Messages;
+
+use ArrayAccess;
+use Countable;
+use Iterator;
+use JsonSerializable;
+
+/**
+ * Class Messages
+ *
+ * Represents a collection of messages
+ */
+class Messages implements ArrayAccess, Countable, Iterator, JsonSerializable
+{
+    /**
+     * @var int
+     */
+    protected int $position = 0;
+
+    /**
+     * @var array
+     */
+    protected array $messages = [];
+
+    /**
+     * Phalcon\Messages\Messages constructor
+     *
+     * @param array $messages
+     */
+    public function __construct(array $messages = [])
+    {
+        $this->messages = $messages;
+    }
+
+    /**
+     * Appends a message to the collection
+     *
+     *```php
+     * $messages->appendMessage(
+     *     new \Phalcon\Messages\Message("This is a message")
+     * );
+     *```
+     * @param MessageInterface $message
+     */
+    public function appendMessage(MessageInterface $message): void
+    {
+        $this->messages[] = $message;
+    }
+
+    /**
+     * Appends an array of messages to the collection
+     *
+     *```php
+     * $messages->appendMessages($messagesArray);
+     *```
+     *
+     * @param MessageInterface[]|Iterator $messages
+     * @throws Exception
+     */
+    public function appendMessages($messages): void
+    {
+        if (!is_iterable($messages)) {
+            throw new Exception("The messages must be iterable");
+        }
+
+        if (is_array($messages)) {
+            /**
+             * An array of messages is simply merged into the current one
+             */
+
+            $this->messages = [...$this->messages, ...$messages];
+        } else {
+            /**
+             * A collection of messages is iterated and appended one-by-one to
+             * the current list
+             */
+
+            $messages->rewind();
+
+            while ($messages->valid()) {
+                $message = $messages->current();
+                $this->appendMessage($message);
+                $messages->next();
+            }
+        }
```

Comment:
I think we need to change this a bit. If the array does not have Message objects then we will store different stuff in the collection. Also, I do not remember why I did the `else` branch that way, most likely to update the `position` property but we can simplify this code - I doubt that this is optimized. 

Thoughts?

Reply:
If the object is iterable, we can foreach and unify the code between the table and the object

Reply:
One could also check the MessageInterface at iteration